### DISCUSSION TITLE
Interrupted / choppy playback on older internet radio

### DISCUSSION THREAD

**Discussion by emtoonst** [AzuraCast/AzuraCast#7147] [Q&A] [answered]
I have an issue with listening to my internet radio station on an Albrecht DR400. It is only on my radiostation, other random radiostations work fine. On other wifi radio players it works fine. The Albrecht is an older device.

[Video of the issue](https://youtu.be/kCsIO8e3JZA?si=BVNbzfKs1cE-Z2sb)

The corresponding log entry in my IceCast log:
[25/May/2024:07:44:24 +0200] "GET /radio.mp3?&_ic2=1 HTTP/1.0" 200 785008 "-" "NSPlayer/11.0.5721.5145" 45

To be honest I do not know how to approach this problem. Do I have to fiddle with the liquidsoap config file?

**Comment by emtoonst** [comment]
I have done some debugging myself. 
Set up a local Icecast server, use Butt to stream to this server and configure the stream url in the internetradio. This works fine, sound is perfect. 

Mountpoint on local Icecast server (working, music is playing):


```
audio_info	ice-bitrate=128;ice-channels=2;ice-samplerate=48000
bitrate	128
genre	various
ice-bitrate	128
ice-channels	2
ice-samplerate	48000
listener_peak	1
listeners	0
listenurl	http://localhost:8000/stream
max_listeners	unlimited
public	0
server_description	Unspecified description
server_name	no name
server_type	audio/mpeg
slow_listeners	0
source_ip	127.0.0.1
stream_start	Sat, 13 Jul 2024 19:48:07 +0200
stream_start_iso8601	2024-07-13T19:48:07+0200
title	Yeah
total_bytes_read	5031600
total_bytes_sent	1068704
user_agent	butt 1.42.0
```

And the mountpoint on my Azuracast (choppy playback):

```
artist	Testartist
audio_codecid	2
audio_info	channels=2;samplerate=44100;bitrate=128
bitrate	128
connected	192215
genre	Rock
incoming_bitrate	130456
listener_connections	2297
listener_peak	3
listeners	0
listenurl	https://**************
max_listeners	-1
metadata_updated	13/Jul/2024:19:54:19 +0200
mpeg_channels	2
mpeg_samplerate	44100
outgoing_kbitrate	0
public	0
queue_size	72725
server_description	Welkom bij de radioomroep 
server_name	Tienhoven
server_type	audio/mpeg
server_url	https://**********
slow_listeners	0
source_ip	127.0.0.1
stream_start	11/Jul/2024:14:30:56 +0200
title	Locus Iste
total_bytes_read	3075437639
total_bytes_sent	339132089
total_mbytes_sent	323
user_agent	Liquidsoap/2.2.5 (Unix; OCaml 4.14.1)
yp_currently_playing	Testartist - Locus Iste
```

Anybody a suggestion?

**Comment by emtoonst** [comment]
Output of curl -s -m 1 -D - -o /dev/null http://****.*******.**/listen/*****_*****_*****_*****/radio.mp3 for the not working stream:

```
HTTP/1.1 200 OK
Server: nginx
Date: Sun, 14 Jul 2024 07:37:39 GMT
Content-Type: audio/mpeg
Transfer-Encoding: chunked
Connection: keep-alive
Pragma: no-cache
Expires: Thu, 19 Nov 1981 08:52:00 GMT
Cache-Control: no-store, no-cache, private
Vary: Origin
icy-logo: https://www.*****-*****-*****.nl/wp-content/uploads/2023/06/Logo-1-kleur-300x213.png
icy-country-code: nl
icy-main-stream-url: https://****.*******.**/listen/*****_*****_*****_*****/radio.mp3
icy-geo-lat-long: 51.957,4.962
icy-br: 128
ice-audio-info: channels=2;samplerate=44100;bitrate=128
icy-description: Welkom bij de radioomroep van de ***** ***** ***** ***** - *****.
icy-genre: Religious
icy-name: ***** ***** ***** *****
icy-pub: 0
icy-url: https://www.*****-a10.nl
X-XSS-Protection: 1
X-Content-Type-Options: nosniff
Referrer-Policy: no-referrer-when-downgrade
```

Output for the working stream: curl -s -m 1 -D - -o /dev/null http://192.168.68.128:8000/stream
```
HTTP/1.0 200 OK
Server: Icecast 2.4.4
Connection: Close
Date: Sun, 14 Jul 2024 07:41:35 GMT
Content-Type: audio/mpeg
Cache-Control: no-cache, no-store
Expires: Mon, 26 Jul 1997 05:00:00 GMT
Pragma: no-cache
Access-Control-Allow-Origin: *
icy-br:128
ice-audio-info: ice-bitrate=128;ice-channels=2;ice-samplerate=48000
icy-br:128
icy-name:no name
icy-pub:0
```

**Reply by emtoonst** [reply]
I have a feeling that it has something to do with Transfer-Encoding: chunked, see [here](https://github.com/kubernetes/ingress-nginx/issues/9988)

**Reply by emtoonst** [reply]
I'll try adding this to nginx.conf:

```
            proxy_http_version 1.0;
            proxy_set_header Connection "";
```

**Comment by BornKillaz** [comment]
I see that there is a sample rate difference and I don't know if this can be somehow related, probably not. 

But, anyway, just giving this a little bump so it can get noticed and probably someone with more knowledge about this can help you.

Good luck. 👍

**Comment by emtoonst** [comment, marked as answer]
Turns out that there is no workaround, streaming server behind Nginx (as with Azuracast docker) will always return Transfer-Encoding: chunked. This will not work with legacy devices.

**Comment by progzilla-radio** [comment]
I have experienced the same problem. And it was resolved by using HSL for legacy devices.

**Reply by emtoonst** [reply]
Thanks, I will give it a try.
